es/es-repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3RepositorySettings.java

```diff
@@ -139,7 +139,6 @@
      */
     static final Setting<String> ENDPOINT_SETTING = Setting.simpleString(
         "endpoint",
-        s -> s.toLowerCase(Locale.ROOT),
```

Comment:
afaik this was ported over from ES in a wrong way. the lowercase function should be the parser and not the validator. so it should be `simpleString("endpoint", "", s -> s.toLowerCase(Locale.Root), ...)`
I'm not sure if the S3 SDK required a lower case endpoint here, but at least ES is doing so..